commit a960c1a0014e0c621522ef3ae1ba07e36642a32d
Author: Thomas Parrott <thomas.parrott@canonical.com>
Date:   Tue Jun 6 15:08:10 2023 +0100

    lxc/copy: Don't try and modify volatile.idmap.next on refresh if not set in source
    
    Fixes #11787
    
    Signed-off-by: Thomas Parrott <thomas.parrott@canonical.com>

diff --git a/lxc/copy.go b/lxc/copy.go
index 5d7ae021b..3a636bd74 100644
--- a/lxc/copy.go
+++ b/lxc/copy.go
@@ -364,7 +364,9 @@ func (c *cmdCopy) copyInstance(conf *config.Config, sourceResource string, destR
 		}
 
 		// Ensure we don't change the target's volatile.idmap.next value.
-		writable.Config["volatile.idmap.next"] = inst.Config["volatile.idmap.next"]
+		if inst.Config["volatile.idmap.next"] != writable.Config["volatile.idmap.next"] {
+			writable.Config["volatile.idmap.next"] = inst.Config["volatile.idmap.next"]
+		}
 
 		op, err := dest.UpdateInstance(destName, writable, etag)
 		if err != nil {
